croquemort/tools.py

```diff
@@ -111,8 +111,50 @@ def retrieve_datetime(datetime_isoformat):
 
 def is_url(url):
     # See https://github.com/kvesteri/validators for reference.
+    ip_middle_oct = u"(?:\.(?:1?\d{1,2}|2[0-4]\d|25[0-5]))"
+    ip_last_oct = u"(?:\.(?:[1-9]\d?|1\d\d|2[0-4]\d|25[0-4]))"
     url_pattern = re.compile(
-        r'^[a-z]+://([^/:]+\.[a-z]{2,10}|([0-9]{{1,3}}\.)'
-        r'{{3}}[0-9]{{1,3}})(:[0-9]+)?(\/.*)?$'
+        u"^"
```

Comment:
That's quite a big change! Is there a [generic test suite](https://github.com/kvesteri/validators/blob/master/tests/test_url.py) that we can borrow somewhere? Or maybe use `validators` as a dependency?

In that case I prefer the dependency.